addon/modifiers/sortable-group.js

```diff
@@ -665,26 +666,56 @@ export default class SortableGroupModifier extends Modifier {
     }
   }
 
+  _getTransitionDuration(item) {
+    const transitionDuration = window.getComputedStyle(item.element).transitionDuration;
+    let time = 0;
+
+    if (transitionDuration) {
+      if (transitionDuration.endsWith('ms')) {
+        time = Number.parseFloat(transitionDuration);
+      } else if (transitionDuration.endsWith('s')) {
+        time = Number.parseFloat(transitionDuration) * 1000;
+      }
+    }
+
+    return Number.isNaN(time) ? 0 : time;
+  }
+
   /**
    * Keeps the UI in sync with actual changes.
    * Needed for drag and keyboard operations.
    */
-  _updateItems() {
+  _updateItems(afterRenderHandler) {
     const items = this.sortedItems;
 
     delete this._firstItemPosition;
 
+    const times = new Map(items.map((item) => [item, this._getTransitionDuration(item)]));
+
     schedule('render', () => {
       items.forEach((item) => item.freeze());
     });
 
     schedule('afterRender', () => {
       items.forEach((item) => item.reset());
-    });
 
-    next(() => {
-      schedule('render', () => {
-        items.forEach((item) => item.thaw());
+      // Trigger the onChange handler after the transforms have been removed in
+      // order to prevent the items to be displayed on incorrect positions (e.g.
+      // left or right from their target locations).
+      afterRenderHandler();
+
+      // Do not restore transition until after it would have been done.
+      // Without this delay, it happens (depending on time) that the reordered
+      // items make a jump before the go to their new location.
+      items.forEach((item) => {
+        const time = times.get(item) ?? 0;
+        later(() => {
+          if (isDestroying(item)) {
+            return;
+          }
+
+          item.thaw();
+        }, time);
```

Comment:
@MathijsPeerlings If this time is too short you can use `time * 1.5` or `time * 2`. Earlier today we used `* 2` in the PoC.